src/http/minreq_http.rs

```diff
@@ -0,0 +1,241 @@
+    basic_auth: Option<String>,
+}
+
+impl Default for MinreqHttpTransport {
+    fn default() -> Self {
+        MinreqHttpTransport {
+            url: format!("localhost:{}", DEFAULT_PORT),
+            timeout: Duration::from_secs(DEFAULT_TIMEOUT_SECONDS),
+            basic_auth: None,
+        }
+    }
+}
+
+impl MinreqHttpTransport {
+    /// Constructs a new [`MinreqHttpTransport`] with default parameters.
+    pub fn new() -> Self {
+        MinreqHttpTransport::default()
+    }
+
+    /// Returns a builder for [`MinreqHttpTransport`].
+    pub fn builder() -> Builder {
+        Builder::new()
+    }
+
+    fn request<R>(&self, req: impl serde::Serialize) -> Result<R, Error>
+    where
+        R: for<'a> serde::de::Deserialize<'a>,
+    {
+        let req = match &self.basic_auth {
+            Some(auth) => minreq::Request::new(minreq::Method::Post, &self.url)
+                .with_timeout(self.timeout.as_secs())
+                .with_header("Authorization", auth)
+                .with_json(&req)?,
+            None => minreq::Request::new(minreq::Method::Post, &self.url)
+                .with_timeout(self.timeout.as_secs())
+                .with_json(&req)?,
+        };
+
+        let resp = req.send()?;
+        let json = resp.json()?;
+        Ok(json)
+    }
+}
+
+impl Transport for MinreqHttpTransport {
+    fn send_request(&self, req: Request) -> Result<Response, crate::Error> {
+        Ok(self.request(req)?)
+    }
+
+    fn send_batch(&self, reqs: &[Request]) -> Result<Vec<Response>, crate::Error> {
+        Ok(self.request(reqs)?)
+    }
+
+    fn fmt_target(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.url)
+    }
+}
+
+/// Builder for simple bitcoind [`MinreqHttpTransport`].
+#[derive(Clone, Debug)]
+pub struct Builder {
+    tp: MinreqHttpTransport,
+}
+
+impl Builder {
+    /// Constructs a new [`Builder`] with default configuration and the URL to use.
+    pub fn new() -> Builder {
+        Builder {
+            tp: MinreqHttpTransport::new(),
+        }
+    }
+
+    /// Sets the timeout after which requests will abort if they aren't finished.
+    pub fn timeout(mut self, timeout: Duration) -> Self {
+        self.tp.timeout = timeout;
+        self
+    }
+
+    /// Sets the URL of the server to the transport.
+    pub fn url(mut self, url: &str) -> Result<Self, Error> {
+        self.tp.url = url.to_owned();
+        Ok(self)
+    }
+
+    /// Adds authentication information to the transport.
+    pub fn basic_auth(mut self, user: String, pass: Option<String>) -> Self {
+        let mut s = user;
+        s.push(':');
+        if let Some(ref pass) = pass {
+            s.push_str(pass.as_ref());
+        }
+        self.tp.basic_auth = Some(format!("Basic {}", &base64::encode(s.as_bytes())));
+        self
+    }
+
+    /// Adds authentication information to the transport using a cookie string ('user:pass').
+    pub fn cookie_auth<S: AsRef<str>>(mut self, cookie: S) -> Self {
```

Comment:
In 9b566579a2ccfd40d95056342ba45b58f14d09be:

You have this function which interprets the data from a cookie file without needing to split it then re-assemble it...